tests/phpunit/tests/blocks/blockHooks.php

```diff
@@ -97,4 +100,82 @@ public function test_get_hooked_blocks_matches_found() {
 			'block hooked at the last child position'
 		);
 	}
+
+	/**
+	 * @ticket XXXXXX
+	 */
+	public function test_inject_hooked_block_at_first_child_position() {
+		$original_markup = <<<HTML
+<!-- wp:tests/group-first-child {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group">
+		<!-- wp:paragraph -->
+			<p>Foo</p>
+		<!-- /wp:paragraph -->
+	</div>
+<!-- /wp:tests/group-first-child -->
+HTML;
+
+		$inserted_block = array(
+			'blockName'    => 'tests/hooked-block',
+			'attrs'        => array(),
+			'innerBlocks'  => array(),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+		);
+
+		$blocks = parse_blocks( $original_markup );
+		$actual = insert_hooked_block( $inserted_block, 'first_child', $blocks[0] );
+
+		$expected_markup = <<<HTML
+<!-- wp:tests/group-first-child {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group">
+		<!-- wp:tests/hooked-block /--><!-- wp:paragraph -->
+			<p>Foo</p>
+		<!-- /wp:paragraph -->
+	</div>
+<!-- /wp:tests/group-first-child -->
+HTML;
+
+		$expected = parse_blocks( $expected_markup )[0];
+		$this->assertSame( $expected, $actual );
+	}
+
+	/**
+	 * @ticket XXXXXX
+	 */
+	public function test_inject_hooked_block_at_last_child_position() {
+		$original_markup = <<<HTML
+<!-- wp:tests/group-last-child {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group">
+		<!-- wp:paragraph -->
+			<p>Foo</p>
+		<!-- /wp:paragraph -->
+	</div>
+<!-- /wp:tests/group-last-child -->
+HTML;
+
+		$inserted_block = array(
+			'blockName'    => 'tests/hooked-block',
+			'attrs'        => array(),
+			'innerBlocks'  => array(),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+		);
+
+		$blocks = parse_blocks( $original_markup );
+		$actual = insert_hooked_block( $inserted_block, 'last_child', $blocks[0] );
+
+		$expected_markup = <<<HTML
+<!-- wp:tests/group-last-child {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group">
+		<!-- wp:paragraph -->
+			<p>Foo</p>
+		<!-- /wp:paragraph --><!-- wp:tests/hooked-block /-->
+	</div>
+<!-- /wp:tests/group-last-child -->
+HTML;
+
+		$expected = parse_blocks( $expected_markup )[0];
+		$this->assertSame( $expected, $actual );
+	}
```

Comment:
These tests might actually be data provider material: We can harmonize them so they all use the same `$original_markup` (i.e. `<!-- wp:tests/group -->` instead of `<!-- wp:tests/group-first-child -->` or `<!-- wp:tests/group-last-child -->`, respectively). The data provider then provides the position, and the corresponding expected resulting markup.

Reply:
Done in e8f5342.